pennylane/_device.py

```diff
@@ -322,6 +322,32 @@ def execute(self, queue, observables, parameters={}, **kwargs):
 
             return self._asarray(results)
 
+    def batch_execute(self, circuits):
+        """Execute a batch of quantum circuits on the device.
+
+        The circuits are represented by tapes, and they are executed one-by-one using the
+        device's ``execute`` method. The results are collected in a list.
+
+        For plugin developers: This function should be overwritten if the device can efficiently run multiple
+        circuits on a backend, for example using parallel and/or asynchronous executions.
+
+        Args:
+            circuits (list[~.tapes.QuantumTape]): circuits to execute on the device
+
+        Returns:
+            list[array[float]]: list of measured value(s)
+        """
+        results = []
+        for circuit in circuits:
+            # we need to reset the device here, else it will
+            # not start the next computation in the zero state
+            self.reset()
```

Comment:
I'm trying to cast my mind back to PL development. I believe it was because there are potential use-cases for executing a circuit on statevector simulators from the _existing_ state, and we didn't want to preclude that? Similar to how the reversible diff method used to work